docs/aws/audit/elbmonitoring/rules/elb_nlb_security_policy.mdx

```diff
@@ -23,6 +23,237 @@ HITRUST, AWSWAF, HIPAA, GDPR, NIST
+     ```sh
+     aws elbv2 create-listener-ssl-policy --name my-secure-policy --protocols TLSv1.2 --ciphers ECDHE-RSA-AES128-GCM-SHA256,ECDHE-RSA-AES256-GCM-SHA384
+     ```
+
+2. **Apply the SSL/TLS Policy to the NLB Listener:**
+   - Identify the ARN of the listener associated with your NLB.
+   - Use the following command to modify the listener to use the secure SSL/TLS policy:
+     ```sh
+     aws elbv2 modify-listener --listener-arn <listener-arn> --ssl-policy my-secure-policy
+     ```
+
+3. **Verify the SSL/TLS Policy:**
+   - Ensure that the SSL/TLS policy has been applied correctly by describing the listener.
+   - Use the following command to describe the listener and verify the SSL/TLS policy:
+     ```sh
+     aws elbv2 describe-listeners --listener-arn <listener-arn>
+     ```
+
+4. **Automate Policy Updates:**
+   - Regularly check for updates to SSL/TLS policies and automate the application of new policies.
+   - Use a script or a scheduled task to periodically update the SSL/TLS policy using the `modify-listener` command as shown above.
+
+By following these steps, you can ensure that your NLBs are always using the latest and most secure SSL/TLS configurations.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Network Load Balancers (NLBs) from having outdated SSL/TLS configurations in AWS Elastic Load Balancer using Python scripts, you can use the Boto3 library, which is the AWS SDK for Python. Here are the steps:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials**:
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables.
+
+3. **Create a Python Script to Update SSL/TLS Configurations**:
+   Write a Python script to check and update the SSL/TLS configurations for your NLBs.
+
+4. **Run the Script**:
+   Execute the script to ensure your NLBs have the latest SSL/TLS configurations.
+
+Here is a sample Python script to achieve this:
+
+```python
+import boto3
+
+# Initialize a session using Amazon EC2
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+
+# Initialize the ELBv2 client
+elbv2_client = session.client('elbv2')
+
+def get_load_balancers():
+    """Retrieve all load balancers"""
+    response = elbv2_client.describe_load_balancers()
+    return response['LoadBalancers']
+
+def update_ssl_tls_configuration(load_balancer_arn):
+    """Update SSL/TLS configuration for a given load balancer"""
+    response = elbv2_client.modify_listener(
+        ListenerArn=load_balancer_arn,
+        SslPolicy='ELBSecurityPolicy-2016-08'  # Example of a modern SSL/TLS policy
+    )
+    return response
+
+def main():
+    load_balancers = get_load_balancers()
+    for lb in load_balancers:
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>